nixos/modules/services/networking/trust-dns.nix

```diff
@@ -0,0 +1,175 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.trust-dns;
+  toml = pkgs.formats.toml { };
+
+  configFile = toml.generate "trust-dns.toml" (
+    lib.filterAttrsRecursive (_: v: v != null) cfg.settings
+  );
+
+  zoneType = lib.types.submodule ({ config, ... }: {
+    options = with lib; {
+      zone = mkOption {
+        type = types.str;
+        description = mdDoc ''
+          Zone name, like "example.com", "localhost", or "0.0.127.in-addr.arpa".
+        '';
+      };
+      zone_type = mkOption {
+        type = types.enum [ "Primary" "Secondary" "Hint" "Forward" ];
+        default = "Primary";
+        description = mdDoc ''
+          One of:
+          - "Primary" (the master, authority for the zone).
```

Comment:
this is very much a noob comment, so feel free to ignore.
Is there somewhere where those are documented so we could add a link there as to what these are and how to make a choice.

Reply:
i grabbed these from [crates/server/src/authority/zone_type.rs](https://github.com/bluejekyll/trust-dns/tree/main/crates/server/src/authority/zone_type.rs#L14). i couldn't find any better documentation in trust-dns for this option. Master/Slave is at least fairly standard jargon in DNS, not sure about the rest... maybe we could link to some docs outside of trust-dns if the others are standard names.

Reply:
alright, the BIND docs explain these options. trust-dns supports only a subset of them, but i think it'll be a good reference to link to regardless: <https://bind9.readthedocs.io/en/v9_18_4/reference.html#type>